File: packages/expo-contacts/android/src/main/java/expo/modules/contacts/models/UrlAddressModel.java
package expo.modules.contacts.models;
import android.database.Cursor;
import android.provider.ContactsContract;

import expo.modules.contacts.EXColumns;

import static android.provider.ContactsContract.*;

public class UrlAddressModel extends BaseModel {
    @Override
    public String getContentType() {
        return CommonDataKinds.Website.CONTENT_ITEM_TYPE;
    }

    @Override
    public String getDataAlias() {
        return "url";
    }

    @Override
    protected String getLabelFromCursor(Cursor cursor) {
        String label = super.getLabelFromCursor(cursor);
        if (label != null)
            return label;
        switch (cursor.getInt(cursor.getColumnIndex(EXColumns.TYPE))) {
        case CommonDataKinds.Website.TYPE_HOME:
            return "home";
        case CommonDataKinds.Website.TYPE_WORK:
            return "work";
        case CommonDataKinds.Website.TYPE_BLOG:
            return "blog";
        case CommonDataKinds.Website.TYPE_FTP:
            return "ftp";
        case CommonDataKinds.Website.TYPE_HOMEPAGE:
            return "homepage";
        case CommonDataKinds.Website.TYPE_PROFILE:
            return "profile";
        case CommonDataKinds.Website.TYPE_OTHER:
            return "other";
        default:
            return "unknown";
        }
    }

}

File: packages/expo-contacts/android/src/main/java/expo/modules/contacts/models/UrlAddressModel.kt
package expo.modules.contacts.models

import android.database.Cursor
import android.provider.ContactsContract.CommonDataKinds
import expo.modules.contacts.Columns

class UrlAddressModel : BaseModel() {
  override val contentType: String = CommonDataKinds.Website.CONTENT_ITEM_TYPE
  override val dataAlias: String = "url"

  override fun getLabelFromCursor(cursor: Cursor): String {
    val label = super.getLabelFromCursor(cursor)
    return label
      ?: when (cursor.getInt(cursor.getColumnIndexOrThrow(Columns.TYPE))) {
        CommonDataKinds.Website.TYPE_HOME -> "home"
        CommonDataKinds.Website.TYPE_WORK -> "work"
        CommonDataKinds.Website.TYPE_BLOG -> "blog"
        CommonDataKinds.Website.TYPE_FTP -> "ftp"
        CommonDataKinds.Website.TYPE_HOMEPAGE -> "homepage"
        CommonDataKinds.Website.TYPE_PROFILE -> "profile"
        CommonDataKinds.Website.TYPE_OTHER -> "other"
        else -> "unknown"
      }
  }
}
